fondant/data_io.py

```diff
@@ -209,7 +207,11 @@ def create_subset_dataframe(
             subset_df = create_subset_dataframe(subset_name, subset_columns, df)
 
             # set index
-            subset_df.set_index("id")
+            if subset_df.index.name != "uid":
+                subset_df["uid"] = (
+                    subset_df["id"].astype("str") + "_" + subset_df["source"]
```

Comment:
General question, do we always expect the users to add an ID + source when creating/adding new subsets? I guess there is no way around it. We just have to make sure it's well documented

Reply:
Well is is loaded in the dataframe we supply to the user so it should already be there for adding columns/subsets for adding rows they will need to add ID's and sources.